dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/ParamEncoder.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+public interface ParamEncoder<T> {
```

Comment:
I wonder if this should extend ListParamEncoder with a default implementation that always returns one value?

Reply:
Do you mind throwing a quick PR together of what you mean? or we can pair